src/main/java/seedu/duke/Penalty.java

```diff
@@ -31,11 +31,32 @@ private static int getUserPenaltyDirection() {
         do {
             System.out.print(PENALTY_PROMPT);
             String directionString = Ui.IN.nextLine().trim();
```

Comment:
I think you should not read the next line here. Next line is already read in Ui class, executeCommand() method.

Reply:
> Next line is already read in Ui class, executeCommand() method.

Correction, next line is read when running Ui.beginListening()

Reply:
Also, as a result of what was mentioned previously, there is no need for a do while loop here, since there is already a loop that will keep listening to user input in Duke.java (the main file).